tests/conftest.py

```diff
@@ -194,6 +198,40 @@ def pytest_collection_modifyitems(session, items):
                 item.add_marker(skip_range)
 
 
+@pytest.fixture(scope="session", name="tmp_dir_cloud")
+def tmp_dir_cloud(cloud_path, cloud):
+    """Create a single client for use by all unit test cases."""
+    ## Read-only filesystem
+    if cloud in READ_ONLY_CLOUDS:
+        yield None
+        return
```

Comment:
This caught my eye as an unusual pattern.  So the desire is to produce `[None]`?

Reply:
Yeah - having some paths `return` and some `yield` was creating issues. I think. This was just moved from elsewhere, and I remember Past Melissa having a good reason for this =P